From 015a738d8548906b7b3bf2561d32a9d7cab5ef8c Mon Sep 17 00:00:00 2001
From: Nicolas Patry <patry.nicolas@protonmail.com>
Date: Thu, 14 Apr 2022 16:29:50 +0200
Subject: [PATCH 1/3] Fixing the documentation for `ByteLevel` in Python (#982)

* Fixing the documentation for `ByteLevel` in Python

* Python stub.py (after rebuilding ofc).
---
 bindings/python/py_src/tokenizers/pre_tokenizers/__init__.pyi | 3 +++
 bindings/python/src/pre_tokenizers.rs                         | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/bindings/python/py_src/tokenizers/pre_tokenizers/__init__.pyi b/bindings/python/py_src/tokenizers/pre_tokenizers/__init__.pyi
index b149ee1..020559a 100644
--- a/bindings/python/py_src/tokenizers/pre_tokenizers/__init__.pyi
+++ b/bindings/python/py_src/tokenizers/pre_tokenizers/__init__.pyi
@@ -100,6 +100,9 @@ class ByteLevel(PreTokenizer):
         add_prefix_space (:obj:`bool`, `optional`, defaults to :obj:`True`):
             Whether to add a space to the first word if there isn't already one. This
             lets us treat `hello` exactly like `say hello`.
+        use_regex (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Set this to :obj:`False` to prevent this `pre_tokenizer` from using
+            the GPT2 specific regexp for spliting on whitespace.
     """
 
     def __init__(self, add_prefix_space=True, use_regex=True):
diff --git a/bindings/python/src/pre_tokenizers.rs b/bindings/python/src/pre_tokenizers.rs
index 5edc6f2..f858ecf 100644
--- a/bindings/python/src/pre_tokenizers.rs
+++ b/bindings/python/src/pre_tokenizers.rs
@@ -228,6 +228,9 @@ macro_rules! setter {
 ///     add_prefix_space (:obj:`bool`, `optional`, defaults to :obj:`True`):
 ///         Whether to add a space to the first word if there isn't already one. This
 ///         lets us treat `hello` exactly like `say hello`.
+///     use_regex (:obj:`bool`, `optional`, defaults to :obj:`True`):
+///         Set this to :obj:`False` to prevent this `pre_tokenizer` from using
+///         the GPT2 specific regexp for spliting on whitespace.
 #[pyclass(extends=PyPreTokenizer, module = "tokenizers.pre_tokenizers", name=ByteLevel)]
 #[text_signature = "(self, add_prefix_space=True, use_regex=True)"]
 pub struct PyByteLevel {}
-- 
2.37.0.windows.1

